### ISSUE TITLE
Validate tokens using JWKS

### ISSUE THREAD

**Issue by simongottschlag** [vouch/vouch-proxy#69]
Hi,

We should add functionality to validate the tokens received from the OpenID Providers. Either add jwks_uri or using the discovery document. I'm not sure they are validated as of now.

**Comment by csckhw303** [comment]
Yes I agree. I don't think it is validating the token after checking existence of token yet.  Is validation function coming up soon?  I appreciate your implementation schedule for it.

**Comment by prateek2408** [comment]
Hello,

Are there any updates on this one, I would be needing this feature as current all JWT tokens are being accepted.

**Comment by bnfinet** [comment]
I see this being implemented after #41 

@prateek2408 are you interested in working on this feature along with #41?  PRs welcome.

**Comment by prateek2408** [comment]
@bnfinet - I can definitely work on this one.

**Comment by bnfinet** [comment]
That's wonderful @prateek2408.  Do let me know if I can be of assistance.

**Comment by bnfinet** [comment]
@simongottschlag I'm going to reopen this one.  I'm pushing on #41 OIDC discovery right now and hope to tackle some aspect of this at the same time.

**Comment by simongottschlag** [comment]
@bnfinet great! 😁👍
